src/YUILoader.cc

```diff
@@ -58,74 +76,93 @@ void YUILoader::loadUI( bool withThreads )
     isGtk = ( ( strstr( envDesktop, "Unity"    ) != NULL ) || isGtk );
     isGtk = ( ( strstr( envDesktop, "XFCE"     ) != NULL ) || isGtk );
 
-    if( isGtk )
+    if( isGtk ) yuiMilestone () << "Detected a Gtk-based desktop environment." << std::endl
+                                << "Prefering Gtk-UI if available and no" << std::endl
+                                << "user-selected override is present." << std::endl;
+
+    YCommandLine cmdline;
+
+    bool wantGtk     = ( cmdline.find( "--gtk" )     != -1 );
+    bool wantNcurses = ( cmdline.find( "--ncurses" ) != -1 );
+    bool wantQt      = ( cmdline.find( "--qt" )      != -1 );
+    bool haveUIPreset = ( wantGtk || wantNcurses || wantQt );
+
+    if( !haveUIPreset )
     {
-	yuiMilestone () << "Detected a Gtk-based desktop environment." << std::endl;
-	yuiMilestone () << "Prefering Gtk-UI if available." << std::endl;
+	wantGtk     = ( strcmp( envPreset, YUIPlugin_Gtk )     == 0 );
+	wantNcurses = ( strcmp( envPreset, YUIPlugin_NCurses ) == 0 );
+	wantQt      = ( strcmp( envPreset, YUIPlugin_Qt )      == 0 );
     }
 
-    YCommandLine cmdline;
+    if( wantGtk )     wantedGUI = YUIPlugin_Gtk;
+    if( wantNcurses ) wantedGUI = YUIPlugin_NCurses;
+    if( wantQt )      wantedGUI = YUIPlugin_Qt;
+
+    yuiMilestone () << "User-selected UI-plugin: \"" << wantedGUI << "\"" << std::endl;
 
-    bool wantNcurses = cmdline.find("--ncurses") != -1;
-    if( wantNcurses )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_NCurses << "\". Forced on command-line." << std::endl;
-    bool wantQt = cmdline.find("--qt") != -1;
-    if( wantQt )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_Qt << "\". Forced on command-line." << std::endl;
-    bool wantGtk = cmdline.find("--gtk") != -1;
-    if( wantGtk )
-	yuiMilestone () << "Using UI-backend: \"" << YUIPlugin_Gtk << "\". Forced on command-line." << std::endl;
+    bool haveGtk     = pluginExists( YUIPlugin_Gtk );
+    bool haveNcurses = pluginExists( YUIPlugin_NCurses );
+    bool haveQt      = pluginExists( YUIPlugin_Qt );
 
-    bool haveQt = pluginExists( YUIPlugin_Qt );
-    bool haveGtk = pluginExists( YUIPlugin_Gtk );
+    wantedGUI="";
```

Comment:
this look like bug in a logic. on line 97-99 you set wantedGUI and here you assign it again to empty.